calico_kubernetes/calico_kubernetes.py

```diff
@@ -326,14 +346,18 @@ def _apply_tags(self, profile_name, pod):
             print('No labels found in pod %s' % pod)
             return
 
+        try:
+            profile = self._datastore_client.get_profile(profile_name)
+        except KeyError:
+            print('Could not find profile with name %s' % profile_name)
+            sys.exit(1)
+
         for k, v in labels.iteritems():
             tag = '%s_%s' % (k, v)
             tag = tag.replace('/', '_')
             print('Adding tag ' + tag)
-            try:
-                self.calicoctl('profile', profile_name, 'tag', 'add', tag)
-            except sh.ErrorReturnCode as e:
-                print('Could not create tag %s.\n%s' % (tag, e))
+            self._datastore_client.profile_update_tags(profile)
```

Comment:
You're missing the line that actually adds the tag to the profile.

Right now, this just gets the profile, then saves the profile :)